### PR TITLE
Multiple little tweaks

### PR DESCRIPTION
**Pull request by korydraughn**

### CODE REVIEW

File: src/endpoints/data_objects/impl.cpp [hunk 1]
```diff
@@ -1605,6 +1605,14 @@ namespace
 					const json input{{"logical_path", lpath_iter->second}, {"options", options}};
 
 					auto conn = irods::get_connection(client_info.username);
+
+					if (!fs::client::is_data_object(conn, lpath_iter->second)) {
```
Comment (alanking) [inline src/endpoints/data_objects/impl.cpp:1611]: Does this prevent touch from creating data objects? If so, should we check for the "no-create" option here?

**Review by alanking — COMMENTED** [review]
I know the tests are WIP, but... still going to suggest tests :)

File: src/endpoints/data_objects/impl.cpp (hunk 1 above)
Comment (korydraughn) [reply src/endpoints/data_objects/impl.cpp:1611]: Good catch! I think a better solution is to use `is_collection` instead.

The intent behind that line is to not allow references to collections.

Will change.

Comment (korydraughn) [reply src/endpoints/data_objects/impl.cpp:1611]: Decided it was better to test for existence and then check if the target is NOT a data object. The current code is only wrong in the case where the path doesn't exist.

Comment (korydraughn) [reply src/endpoints/data_objects/impl.cpp]: Done. Please review and let me know if anything sticks out to you. Thanks.

**Comment by korydraughn** [comment]
Added tests. Everything is passing.

**Review by alanking — COMMENTED** [review]
Very nice. I say squash it once any open conversations are resolved.

File: test/test_irods_http_api.py [hunk 2]
```diff
@@ -582,6 +579,146 @@ def test_common_operations(self):
...
+                'lpath': data_object,
+                'ppath': physical_path,
+                'resource': other_resource,
+                'as-additional-replica': 1
+            })
+            #print(r.content) # Debug
+            self.assertEqual(r.status_code, 200)
+            self.assertEqual(r.json()['irods_response']['error_code'], 0)
+
+            # Show a new data object exists with the expected replica information.
```
[the remark this answers is not in the document]
Comment (korydraughn) [reply test/test_irods_http_api.py]: Done.

**Comment by korydraughn** [comment]
Ignore clang-format. It's producing the wrong result for some odd reason.

All tests pass. Adding pounds.

**Review by alanking — APPROVED** [review]

**Comment by korydraughn** [comment]
#'d

**Pull request merged**